command/sidechain/registration/register_validator.go

```diff
@@ -0,0 +1,236 @@
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
+	"github.com/0xPolygon/polygon-edge/contracts"
+	"github.com/0xPolygon/polygon-edge/secrets"
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/spf13/cobra"
+	"github.com/umbracle/ethgo"
+)
+
+var (
+	stakeManager         = contracts.ValidatorSetContract
+	stakeFn              = contractsapi.ChildValidatorSet.Abi.Methods["stake"]
+	registerFn           = contractsapi.ChildValidatorSet.Abi.Methods["register"]
+	newValidatorEventABI = contractsapi.ChildValidatorSet.Abi.Events["NewValidator"]
+	stakeEventABI        = contractsapi.ChildValidatorSet.Abi.Events["Staked"]
+)
+
+var params registerParams
+
+func GetCommand() *cobra.Command {
+	registerCmd := &cobra.Command{
+		Use:     "register-validator",
+		Short:   "Registers and stake an enlisted validator",
+		PreRunE: runPreRun,
+		RunE:    runCommand,
+	}
+
+	setFlags(registerCmd)
+
+	return registerCmd
+}
+
+func setFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(
+		&params.accountDir,
+		polybftsecrets.DataPathFlag,
+		"",
+		polybftsecrets.DataPathFlagDesc,
+	)
+
+	cmd.Flags().StringVar(
+		&params.configPath,
+		polybftsecrets.ConfigFlag,
+		"",
+		polybftsecrets.ConfigFlagDesc,
+	)
+	cmd.Flags().StringVar(
+		&params.stake,
+		stakeFlag,
+		"",
+		"stake represents amount which is going to be staked by the new validator account",
+	)
+	cmd.Flags().Int64Var(
+		&params.chainID,
+		chainIDFlag,
+		command.DefaultChainID,
+		"the ID of the chain",
+	)
+
+	helper.RegisterJSONRPCFlag(cmd)
+}
+
+func runPreRun(cmd *cobra.Command, _ []string) error {
+	params.jsonRPC = helper.GetJSONRPCAddress(cmd)
+
+	return params.validateFlags()
+}
+
+func runCommand(cmd *cobra.Command, _ []string) error {
+	outputter := command.InitializeOutputter(cmd)
+	defer outputter.WriteOutput()
+
+	secretsManager, err := polybftsecrets.GetSecretsManager(params.accountDir, params.configPath, true)
+	if err != nil {
+		return err
+	}
+
+	txRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithIPAddress(params.jsonRPC))
+
+	newValidatorAccount, err := wallet.NewAccountFromSecret(secretsManager)
+	if err != nil {
+		return err
+	}
+
+	sRaw, err := secretsManager.GetSecret(secrets.ValidatorBLSSignature)
+	if err != nil {
+		return err
+	}
+
+	sb, err := hex.DecodeString(string(sRaw))
+	if err != nil {
+		return err
+	}
+
+	blsSignature, err := bls.UnmarshalSignature(sb)
+	if err != nil {
+		return err
+	}
+
+	receipt, err := registerValidator(txRelayer, newValidatorAccount, blsSignature)
+	if err != nil {
+		return err
+	}
+
+	if receipt.Status != uint64(types.ReceiptSuccess) {
+		return errors.New("register validator transaction failed")
+	}
+
+	result := &registerResult{}
+	foundLog := false
+
+	for _, log := range receipt.Logs {
```

Comment:
Maybe we should have helper function which receives slice of logs and match it with another parameter and if matching was successful returns parsed event (otherwise nil or error)?
We have similar loop on two places and potentially on few more